sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala

```diff
@@ -94,11 +97,18 @@ class ResolveCatalogs(val catalogManager: CatalogManager)
           s"because view support in catalog has not been implemented yet")
 
     case AlterNamespaceSetPropertiesStatement(NonSessionCatalog(catalog, nameParts), properties) =>
-      AlterNamespaceSetProperties(catalog.asNamespaceCatalog, nameParts, properties)
+      val availableProps = properties -- RESERVED_PROPERTIES.asScala
```

Comment:
I think it's better to fail if reserved properties are being set/altered.

Reply:
ok, I will go this way.

Reply:
I notice that the `RESERVED_PROPERTIES` is unremovable but not reversed.. the `comment` and `location` is not exactly same with `ownerName/Type`. the name `RESERVED_PROPERTIES ` here is deceptive.

Reply:
can Hive alter table location via ALTER TABLE SET TBLPROPERTIES?

Seems like Hive can only do it for table comment: https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-AlterTableComment

Reply:
Hive will not change Database field members via properties, if we set dbproperites('location'='xxx'), the `location` here is just a property inside properties, meaningless to the catalog

Reply:
OK so we can't follow Hive here. It seems wrong to me to change location and owner with SET TBLPROPERTIES or SET DBPROPERTIES, but comment should be fine.